core/src/main/java/io/jenkins/pluginhealth/scoring/probes/JiraOpenIssuesProbe.java

```diff
@@ -0,0 +1,133 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.jenkins.pluginhealth.scoring.probes;
+
+import static java.time.temporal.ChronoUnit.SECONDS;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.util.Optional;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+@Component
+@Order(AbstractOpenIssuesProbe.ORDER)
+class JiraOpenIssuesProbe extends AbstractOpenIssuesProbe {
+    public static final String KEY = "jira-open-issues";
+    private static final String JIRA_HOST = "https://issues.jenkins.io/rest/api/latest/search?";
+    private static final Logger LOGGER = LoggerFactory.getLogger(JiraOpenIssuesProbe.class);
+    final ObjectMapper objectMapper = new ObjectMapper();
+    HttpClient httpClient;
+    HttpRequest httpRequest;
+
+    /**
+     * Get total number of open JIRA issues in a plugin.
+     *
+     * @param context {@link ProbeContext}
+     * @return Optional of type Integer.
+     */
+    @Override
+    Optional<Integer> getCountOfOpenIssues(ProbeContext context) {
+        if (context.getIssueTrackerUrlsByNames() == null) {
+            LOGGER.info("IssueTracker has no JIRA open issues for the plugin.");
+            return Optional.empty();
+        }
+
+        /* Stores the JIRA URL to view all existing issues in the plugin. Ex: https://github.com/jenkinsci/cloudevents-plugin/issues */
+        String viewJiraIssuesUrl = context.getIssueTrackerUrlsByNames().get("jira");
+
+        if (viewJiraIssuesUrl == null || viewJiraIssuesUrl.isEmpty()) {
+            LOGGER.info("The plugin does not use JIRA to track issues.");
+            return Optional.empty();
+        }
+        try {
+            /* The `url` will contain the JIRA url to view issues.
+               For ex: https://issues.jenkins.io/rest/api/latest/search?jql=component=15979
+            */
+            URL url = new URL(viewJiraIssuesUrl);
```

Comment:
Because `URL#equals` method is doing a DNS resolution, I'd prefer using `URI` class which provide the same features but with less problems.